src/spv/inst-make.cpp

```diff
@@ -1210,6 +1210,29 @@ bool Instruction::makeResult(
     case spv::OpLabel: // 248
         data[result_at].redefine(new Primitive(location));
         break;
+    case spv::OpTypeRayQueryKHR: { // 4472
+        data[result_at].redefine(new Type(Type::rayQuery()));
+        break;
+    }
+    case spv::OpTypeAccelerationStructureKHR: // 5341
+        // Will define the type in class "AccelerationStructureManager" in "acceleration-structure.cxx" in "copyFrom()"
```

Comment:
I think this comment can be improved. I remember you explaining this concept, but I don't remember all the details. Correct me if I am wrong, but from what I recall, the reason why we cannot define the type here is because the SPIR-V instruction doesn't give enough info (many fields of raytracing types are opaque), so we have to wait until a use. Right?

If that is the case, it may be wise to create a new pseudo constructor for acceleration structure types which doesn't take any arguments. It seems silly to me to pass in empty vectors here. Just let the type populate them itself.

Reply:
Correct! I've decided to use default arguments for creating an empty acceleration structure type.